### DISCUSSION TITLE
Bible book name component / Bible reference component

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#304] [Development]
It would be useful to have a custom web component for the book names.
```html
<bible-bookname>Matt</bible-bookname>
```
would be transformed to 
```html
<span>Matthew</span>
```

```html
<bible-bookname lang="ru" as="h1">Matt</bible-bookname>
```
would be transformed to 
```html
<h1>Матфея</h1>
```

In this case, developers don't have to worry about keeping and updating localized book names.

This can be farther extended to as Bible reference component with different versification systems and reference separators: 
```html
<bible-ref lang="fr">Matt 5:3-12</bible-bookname>
```
would be transformed to 
```html
<span>Matthieu 5,3-12</span>
```

These components should be completely independent to be able to use them outside of Ezra Bible App

**Comment by tobias-klein** [comment]
I like your idea! Currently we depend on SWORD plus the internal IPC communication for transforming book codes into localized book names. If you want to use this outside of _Ezra Bible App_ the question is whether the SWORD dependency is useful for that ... are you thinking of a pure JavaScript version? What could possibly be done is at least use the locales.d from SWORD as an input and parse that directly from JavaScript ... but then even for that you need either node.js / file system access or you need to write a script that turns the locales.d files into JS files.

**Comment by zhuiks** [comment]
The main idea is to have zero dependencies. Obviously it should be based on SWORD `locales.d` as the main source of data. So that might be a dev dependency. And files can be parsed to json during build process. I have to experiment/think through on how to pack it and see what the bundle size will be in the end.
